pages/api/v1/users/[username]/index.public.js

```diff
@@ -112,7 +112,8 @@ async function patchHandler(request, response) {
   try {
     await transaction.query('BEGIN');
 
-    const updatedUser = await user.update(targetUsername, secureInputValues, {
+    const updatedUser = await user.update(targetUser.id, secureInputValues, {
+      oldUser: targetUser,
```

Comment:
Ao invés aceitar o `id` ou `username`, que tal aceitar um `targetUser`? Assim não precisa de `options.oldUser`, mas mantém as otimizações feitas em `user.update`.

E deve ser possível fazer uma outra otimização, que é só buscar o `targetUser` no banco se o `targetUsername` for diferente de `userTryingToPatch.username`.

Reply:
Acho que as duas sugestões juntas ficaram confusas.

Você diz para a interface da função ser assim?

```js
async function update(targetUser, postedUserData, options = {})
```

E o `user.update` precisaria tomar o cuidado do objeto estar incompleto? Por exemplo, algo como:

```diff
-  const currentUser = options.oldUser ?? (await findOneById(userId, { transaction: options.transaction }));
+  const currentUser = 'username' in targetUser ? targetUser : (await findOneById(userId, { transaction: options.transaction }));
```

(ou alguma verificação melhor)

Reply:
Rafa, perdão pela falta de clareza. Tudo que você escreveu se refere à sugestão 1, que é apenas uma refatoração para evitar passar `targetUser.id` e `targetUser` para a mesma função. Inclusive concordo que **talvez** dê para fazer uma verificação melhor.

---

Já a sugestão 2, essa sim é uma otimização extra, que é neste trecho abaixo, onde você não mexeu:

https://github.com/filipedeschamps/tabnews.com.br/blob/114679cfe8e1dc7dfcb5c6fc5390c8f877115847/pages/api/v1/users/%5Busername%5D/index.public.js#L79-L81

A não ser quando é um moderador editando outro usuário, essa busca pelo `targetUser` não é necessária, já que é o mesmo que `userTryingToPatch`. Por isso falei em comparar os dois `usernames`.

Reply:
Deixei essa alteração no último commit (https://github.com/filipedeschamps/tabnews.com.br/pull/1737/commits/bd3fd2b5322221f69db4b1818ae91f185c711bce), para ficar mais claro o que foi modificado.

Simplifiquei o máximo que consegui.